### PR TITLE
[doc] clarify lazy-eval of removeObjects

### PR DESCRIPTION
**Pull request by KevinSmile**
as discussed in https://github.com/minio/minio-java/issues/810

### CODE REVIEW

File: api/src/main/java/io/minio/MinioClient.java [hunk 1]
```diff
@@ -2987,19 +2987,27 @@ public void removeObject(String bucketName, String objectName)
   /**
    * Removes multiple objects from a bucket.
    *
+   * <p><b>Note:</b></p>
+   * This method does lazy evaluation. You need to traverse over the returned Iterable to do the
+   * real delete jobs.
```
Comment (balamurugana) [inline api/src/main/java/io/minio/MinioClient.java]: You would need to modify description than adding new style here. Below is an idea.
```
Removes multiple objects from a bucket. As objects removal are lazily executed, its required to iterate returned Iterable.
```

File: api/src/main/java/io/minio/MinioClient.java [hunk 2]
```diff
@@ -2987,19 +2987,27 @@ public void removeObject(String bucketName, String objectName)
...
+   *
    * </p><b>Example:</b><br>
    * <pre>{@code // Create object list for removal.
    * List<String> objectNames = new LinkedList<String>();
    * objectNames.add("my-objectname1");
    * objectNames.add("my-objectname2");
    * objectNames.add("my-objectname3");
+   * // need to traverse over the returned Iterable to do the real delete jobs.
```
Comment (balamurugana) [inline api/src/main/java/io/minio/MinioClient.java]: I don't think this comment is needed as the javadoc is clear to say about lazy eval.

File: docs/API.md [hunk 3]
```diff
@@ -3079,7 +3079,7 @@ __Parameters__
 
 |Return Type	  | Exceptions	  |
 |:--- |:--- |
-| ``Iterable<Result<DeleteError>>``:an iterator of Result DeleteError.  | _None_  |
+| ``Iterable<Result<DeleteError>>``:a lazy iterator of Result DeleteError.  | _None_  |
```
Comment (balamurugana) [inline docs/API.md]: I don't think this change is needed as the description should carry about lay eval.

File: docs/API.md [hunk 4]
```diff
@@ -3093,6 +3093,7 @@ objectNames.add("my-objectname2");
 objectNames.add("my-objectname3");
 try {
       // Remove object all objects in objectNames list from the bucket my-bucketname.
+      // Lazy evaluation. Need to traverse over the returned Iterable to do the real delete jobs.
```
Comment (balamurugana) [inline docs/API.md]: I don't think this comment is needed as the description carries about lay eval.

File: examples/RemoveObjects.java [hunk 5]
```diff
@@ -48,6 +48,7 @@ public static void main(String[] args)
       objectNames.add("my-objectname3");
 
       // Remove object all objects 'objectNames' list from 'my-bucketname'.
+      // Lazy evaluation. Need to traverse over the returned Iterable to do the real delete jobs.
```
Comment (balamurugana) [inline examples/RemoveObjects.java]: You could add comment like `Its required to traverse over the returned Iterable for lazy evaluation`

File: api/src/main/java/io/minio/MinioClient.java (hunk 1 above)
Comment (KevinSmile) [reply api/src/main/java/io/minio/MinioClient.java]: ok

**Comment by balamurugana** [comment]
@KevinSmile you would need to check travis-ci failures and fix them accordingly.

**Pull request merged**